spacy/ml/extract_spans.py

```diff
@@ -0,0 +1,52 @@
+from typing import Tuple, Callable
+from thinc.api import Model
+from thinc.types import Ragged, Ints1d
+
+
+def extract_spans() -> Model[Tuple[Ragged, Ragged], Ragged]:
```

Comment:
Register this layer?

Reply:
For 'legacy' purposes later on, or do you want to have an explicit reference to an 'extract spans' layer in the config of `spacy.SpanCategorizer.v1` ? I'm not sure it would make much sense to overwrite this functionality?

Reply:
No, I think our policy is that the active config should use the import, so this is just adding it so it's there later for legacy purposes.

Reply:
okay. Actually, looking at this, now I'm thinking that `spacy.extract_spans.v1` and `spacy.mean_max_reducer.v1` should be `layers` instead of `architectures`? Not sure we have a crisp definition to distinguish between the two.

Reply:
Oh, I hadn't looked at that in detail elsewhere. I thought this would be a `layer`, but you're probably right that the distinction isn't clear.

Reply:
Yea I think we tend to use `architectures` if they are the top thing used by a `model`.

Reply:
I think `layers` is the right choice here, just like for the other layers in `spacy/ml`?

Reply:
Yea, I agree, changed them to `layers`